docs/Swatches.md

```diff
@@ -0,0 +1,159 @@
+
+Banners that are running on skins that don't have a dark mode toggle can include the `skin_default.scss` implementation. The variables are declared inside the `.wmde-banner` selector.
+
+Banners that are running on the Vector 2022 skin can include the `skin_vector-2022.scss`. The variables in this are declared 4 times:
+
+1. When the donor has selected *Light Mode*.
+2. When the donor has selected *Dark Mode*.
+3. When the donor has selected *Automatic* and their OS is set to *dark mode*.
+4. When the donor has selected *Automatic* and their OS is set to *light mode*.
+
+If/when another skin starts to implement dark mode we will need to create a new implementation for that.
+
+### Optional Variables
+
+Because all banners don't include all components/features, we define [default scss variables](https://sass-lang.com/documentation/at-rules/use/#configuration) as boolean flags in the swatch implementations that allow you to include only the variables for the features the banner uses. This helps keep the compiled css a little smaller, especially on banners that need to declare the variables 4 times.
+
+Whenever you add a new component/feature, you'll have to add a section to `colors_light.scss` and `colors_dark.scss` where you define the new color names:
+
+```scss
+$grey150: #eeeeee;
+
+@mixin swatch(
+    // ...
+    $myNewFeature: false,
+    // ...
+) {
+    @if $myNewFeature {
+        --my-new-feature-background: #{$grey150};
+        --my-new-feature-border: #{$grey150};
+    }
+}
+```
+
+You also need to add the flag to the skin file:
+
+```scss
+$myNewFeature: false !default;
+
+.wmde-banner {
+    @include light.swatch(
+        // ...
+        $myNewFeature,
+        // ...
+    );
+}
+
+```
+
+### Using Swatches
+
+You import the swatch for the skin the banner is running on at the top of the banner's `style.scss`. Then add the flags for the features/components used in your banners, e.g.:
+
+```scss
+@use 'src/themes/Treedip/swatches/skin_default' with (
+    $slider: true,
+    $progress-bar: true,
+    $select-group-button: true,
+    $upgrade-to-yearly: true,
+    $fallback-banner: true,
+);
+```
+
+Not all banner themes use swatches yet, but all banners will need to include the Use of Funds swatches:
+
+```scss
+@use 'src/themes/UseOfFunds/swatches/skin_default' as uof-default;
+```
+
+Please note the use of namespacing here as the compiler will complain if we use 2 modules with the same name and default is already defined when importing the main swatch.
+
+### Converting a theme to a swatch:
+
+When you need to convert an existing theme to use the swatches you should follow these steps:
+
+1. Create `color_light.scss` and `skin_default.scss` files in a swatches folder.
+2. Copy the colours from the `variables/_colors.scss` file into the `color_light.scss` and delete `variables/_colors.scss`. Deleting the old file first causes the compiler to throw errors which makes it a little easier to find all the usages.
```

Comment:
>[!Question]
> Do we have a strategy on how to create/deduct the colors for a `color_dark.scss` file yet?
> Or do we basically just leave that to the designer, providing us with a dark design + colors via Figma?

Reply:
This is another question for the PM. I can do the swatch work myself, but I'm in limbo about how much UX availability we have and if I can just go ahead and do it or not.

If I was to do it I'd do it straight in code, no Figma needed.